livekit-plugins/livekit-plugins-google/livekit/plugins/google/beta/realtime/realtime_api.py

```diff
@@ -271,25 +273,37 @@ def update_options(
         if is_given(temperature):
             self._opts.temperature = temperature
 
-        if self._session:
-            logger.warning("Updating options; triggering Gemini session reconnect.")
-            self._reconnect_event.set()
-            self._schedule_gemini_session_close()
+        self.reset_gemini_session()
 
     async def update_instructions(self, instructions: str) -> None:
         self._opts.instructions = instructions
-        if self._session:
-            logger.warning("Updating instructions; triggering Gemini session reconnect.")
-            self._reconnect_event.set()
-            self._schedule_gemini_session_close()
+        self.reset_gemini_session()
 
     async def update_chat_ctx(self, chat_ctx: llm.ChatContext) -> None:
         async with self._update_chat_ctx_lock:
-            self._chat_ctx = chat_ctx
-            turns, _ = to_chat_ctx(self._chat_ctx, id(self), ignore_functions=True)
-            tool_results = get_tool_results_for_realtime(self._chat_ctx)
-            if turns:
-                self._msg_ch.send_nowait(LiveClientContent(turns=turns, turn_complete=False))
+            diff = llm.utils.compute_chat_ctx_diff(self._chat_ctx, chat_ctx)
+            if diff.to_remove:
+                self._chat_ctx = chat_ctx
+                self.reset_gemini_session()
```

Comment:
I think it's OK to reconnect, but ideally, we use aiohttp with a connection pool